src/NetMQ/ReceivingSocketExtensions.cs

```diff
@@ -393,13 +370,9 @@ public static void ReceiveMessage([NotNull] this IReceivingSocket socket, [NotNu
         {
             message.Clear();
 
-            bool more = true;
-
+            var more = true;
             while (more)
-            {
-                byte[] buffer = socket.Receive(dontWait, out more);
-                message.Append(buffer);
-            }
+                message.Append(socket.Receive(dontWait, out more));
```

Comment:
I also think we should make the NetMQMessage disposable (including frames) and then use the Msg inside the NetMQFrame, it will save copy and allocation. But this will change the behavior of NetMQMessage a little, so we need to think how to do that (maybe new class). For example now after sending the message the message will be cleaned (we don't own the msg after sending) which is different behavior.

I believe we can make Message class with good performance (minimal allocation and copying) which be very good alternative to send/recv methods.